core/src/commonTest/kotlin/com/xebia/functional/xef/vectorstores/LocalVectorStoreSpec.kt

```diff
@@ -0,0 +1,34 @@
+package com.xebia.functional.xef.vectorstores
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class LocalVectorStoreSpec :
+    StringSpec({
+        "memories function should return all of messages in the right order when the limit is greater than the number of stored messages" {
+            val localVectorStore = LocalVectorStore(FakeEmbeddings())
+
+            localVectorStore.addMemories(messages1)
+            localVectorStore.addMemories(messages2)
+
+            val messages = localVectorStore.memories(defaultConversationId, Int.MAX_VALUE)
+
+            val messagesExpected = messages1 + messages2
+
+            messages shouldBe messagesExpected
+        }
+
+        "memories function should return the last n messages in the right order" {
+            val localVectorStore = LocalVectorStore(FakeEmbeddings())
+
+            localVectorStore.addMemories(messages1)
+            localVectorStore.addMemories(messages2)
+
+            val messages = localVectorStore.memories(defaultConversationId, 2)
```

Comment:
I think I would define a value for that `2` you are using as the `limit` to be used also in the `takeLast` function